li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/largemessage/ConsumerRecordsProcessor.java

```diff
@@ -417,6 +423,24 @@ public void close(long timeout, TimeUnit timeUnit) {
       }
 
       _partitionConsumerHighWatermark.computeIfAbsent(tp, _storedConsumerHighWatermark)._currentConsumerHighWatermark = consumerRecord.offset();
+      LargeMessageHeaderValue largeMessageHeaderValue = null;
+      Headers headers = new RecordHeaders(consumerRecord.headers());
+      Iterator<Header> iterable = headers.headers(Constants.LARGE_MESSAGE).iterator();
+      if (iterable.hasNext()) {
+        Header header = iterable.next();
+        largeMessageHeaderValue = LargeMessageHeaderValue.fromBytes(header.value());
+      }
+      // Once the large message is parsed, remove any such key from record headers
+      headers.remove(Constants.LARGE_MESSAGE);
```

Comment:
I think lines 434-443 can actually be put inside the if block in line 429-432?

Reply:
I have simplified the code. What do you think now?